image-resize/function.js

```diff
@@ -0,0 +1,59 @@
+const cp = require('child_process');
+const request = require('request-promise-native');
+const file = require('./file');
+
+exports.handler = async (body, context) => {
+
+  // Get an image URL from the request body or query
+  // string (useful for invoking directly from a browser)
+  const uri = body.url || context.request.query.url;
+  if (!uri) {
+    // This will be caught by Binaris and returned as
+    // a HTTP 500 status code
+  	throw new Error('No URL');
+  }
+
+  // Get a target size for the image. Image will be
+  // resized to {size} by {size} pixels. If neither
+  // request body nor query string contains a size
+  // value then default to 32
+  const size = parseInt(body.sz) || parseInt(context.request.query.sz) || 32;
```

Comment:
Also why do you not check with an `if` statement like you did for `uri`.

I recommend creating a separate helper function which takes in the body, QP's, desired field to extract, and default. That way you can just extract.

Here is the code I wrote for dynamo example

```
function validatedBody(body, ...fields) {
  for (const field of fields) {
    if (!Object.prototype.hasOwnProperty.call(body, field)) {
      throw new Error(`Missing request body parameter: ${field}.`);
    }
  }
  return body;
}
```

And used like

```
  const {
    driverID,
    rideStatus,
    lastLocation
  } = validatedBody(body, 'driverID', 'rideStatus', 'lastLocation');
```

Clearly it needs some modification to support QP.